docker/Dockerfile

```diff
@@ -11,7 +11,7 @@
 ### the date in the FROM command
 ### Date format is ISO year/month/day
 FROM archlinux:base-20210110.0.13332
-RUN echo "Server=https://archive.archlinux.org/repos/2021/01/10/\$repo/os/\$arch" | tee /etc/pacman.d/mirrorlist
+RUN echo -e "Server=http://172.17.0.1:7878/\$repo/os/\$arch\nServer=https://archive.archlinux.org/repos/2021/01/10/\$repo/os/\$arch" | tee /etc/pacman.d/mirrorlist
```

Comment:
Is this IP guaranteed to always be this?

Reply:
Yeah it is, during our build, the buildx container defaults to the `bridge` network driver, from which you can always ping the host on `172.17.0.1`

Alternatively, we could add `--add-host host.docker.internal:host-gateway` to [here](https://github.com/NUbots/NUbots/blob/master/tools/utility/dockerise/platform.py#L87), and then instead of `172.17.0.1` it could be `host.docker.internal`, thoughts?

Reply:
The only alternative I can think of is to set an environment variable to the output of this bash command
```
export DOCKER_GATEWAY=$(ip -o addr show | grep inet\  | grep "docker" | cut -d\  -f 7 | cut -d'/' -f1)
```

But this also assumes that the interface will always have `docker` in its name. But, if the `--add-host` command works then I would go with that option.

Reply:
I couldn't get `--add-host host.docker.internal:host-gateway` to work with `buildx`, it didnt seem to recognise `host-gateway`, it does work with `build` though...

I ended up using docker inspect to get the docker host ip, and then just passed that too `--add-host`

https://github.com/NUbots/NUbots/pull/390/files#diff-e8c882de7db134796dc7fe8e647cae033d8faa7d34027e24b451ecf25d7487c0R86